src/app.ts

```diff
@@ -13,14 +17,102 @@ interface TemperatureSummary {
   average: number
 }
 
+class dataByTime implements TemperatureReading {
+  public dataTemperature: Array<number> = []
+  // I created a new Class by error not definition
+  public TSummary: ClassTSummary = new ClassTSummary(0)
+  constructor(
+    public time: Date,
+    public city: string,
+    public temperature: number,
+  ) {
+    this.dataTemperature.push(temperature)
+  }
+
+  addTmpe(num: number): void {
+    this.dataTemperature.push(num)
+  }
+}
+
+class ClassTSummary implements TemperatureSummary {
+  constructor(temperature: number) {
+    this.first = this.last = this.high = this.low = this.average = temperature
+  }
+  first
+  last
+  high
+  low
+  average
+}
+
+function existValue(data: Date) {
+  for (const d of allData) {
+    if (d.time.getTime() === data.getTime()) return true
+  }
+
+  return false
+}
+
+//resolving y updating sumary
+function getSummary(tmpSummary: ClassTSummary, value: number): ClassTSummary {
+  if (tmpSummary.first === -1) tmpSummary.first = value
+  if (tmpSummary.high < value) tmpSummary.high = value
+
+  if (tmpSummary.low > value) tmpSummary.low = value
+
+  tmpSummary.last = value
+  tmpSummary.average += value
+  return tmpSummary
+}
```

Comment:
it's generally considered bad practice to mutate the parameters you pass in. In this case you do, but you won't always know who "owns" the data, where it came from, or what it's meant to do, so it's best to avoid mutating data altogether. The better approach would be to create a new object here and return it.